curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/6434759f78ec812264ff8f34.md

```diff
@@ -49,6 +49,12 @@ Your callback function should be empty.
 assert.match(code, /letters\.forEach\(\s*\(?\s*letter\s*\)?\s*=>\s*\{\s*\}/)
```

Comment:
```suggestion
assert.match(code, /letters\.forEach\(\s*(\(\s*letter\s*\)|letter)\s*=>\s*\{\s*\}/)
```
While we are here...

Reply:
Are there valid patterns the existing regex doesn't match, but this one does ? Just curious.

Reply:
It's the other way around, current will match invalid code, ie.:
```javascript
letters.forEach(letter) => {}
```

Reply:
Got it 👍